jvb/src/main/kotlin/org/jitsi/videobridge/SsrcCache.kt

```diff
@@ -0,0 +1,616 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge
+
+import org.jitsi.nlj.MediaSourceDesc
+import org.jitsi.nlj.VideoType
+import org.jitsi.nlj.rtp.SsrcAssociationType
+import org.jitsi.nlj.rtp.codec.vp8.Vp8Packet
+import org.jitsi.nlj.rtp.codec.vp9.Vp9Packet
+import org.jitsi.rtp.rtcp.RtcpPacket
+import org.jitsi.rtp.rtcp.RtcpSrPacket
+import org.jitsi.rtp.rtcp.rtcpfb.RtcpFbPacket
+import org.jitsi.rtp.rtp.RtpPacket
+import org.jitsi.rtp.util.RtpUtils
+import org.jitsi.utils.LRUCache
+import org.jitsi.utils.MediaType
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.logging2.createChildLogger
+import org.jitsi.videobridge.message.AudioSourceMapping
+import org.jitsi.videobridge.message.AudioSourcesMap
+import org.jitsi.videobridge.message.BridgeChannelMessage
+import org.jitsi.videobridge.message.VideoSourceMapping
+import org.jitsi.videobridge.message.VideoSourcesMap
+import org.jitsi.videobridge.relay.AudioSourceDesc
+
+/**
+ * Get tl0PicIdx field for codecs that have it.
+ * Return -1 if not applicable.
+ */
+private fun RtpPacket.getTl0Index(): Int {
+    return when (this) {
+        is Vp9Packet -> this.TL0PICIDX
+        is Vp8Packet -> this.TL0PICIDX
+        else -> -1
+    }
+}
+
+/**
+ * Set tl0PicIdx field for codecs that have it.
+ * No-op if not applicable.
+ */
+private fun RtpPacket.setTl0Index(tl0Index: Int) {
+    when (this) {
+        is Vp9Packet -> this.TL0PICIDX = tl0Index
+        is Vp8Packet -> this.TL0PICIDX = tl0Index
+    }
+}
+
+/**
+ * Truncate to 8 unsigned bits.
+ */
+private fun Int.fixByte(): Int {
+    val s = this.rem(256)
+    /* Adjust if dividend was negative. */
+    return if (s < 0) (s + 256) else s
+}
```

Comment:
Shouldn't `this and 0xff` do this?  Bitwise operations are defined on signed integers in Java/Kotlin.

Reply:
done, thanks.